src/main/java/org/broadinstitute/hellbender/tools/copynumber/GermlineCNVCaller.java

```diff
@@ -0,0 +1,408 @@
+ * library preparation and sequencing protocols. For WES and WGS, we recommend at least 30 samples for the cohort
+ * mode.</p>
+ *
+ * <p>The parametrized coverage model can be used for CNV detection on future case samples provided that they are
+ * strictly compatible (in terms of library preparation and sequencing protocol) with the cohort used to generate the
+ * model parameters. We refer to this run-mode as the <b>CASE</b> mode. There is no lower limit on the number of
+ * samples for running {@link GermlineCNVCaller} in the case mode.</p>
+ *
+ * <p>In both run-modes, the output calls of {@link DetermineGermlineContigPloidy} are required for all samples.
+ * The germline contig ploidy estimates are used for choosing the baseline copy-number state (in particular, for
+ * sex chromosomes).</p>
+ *
+ * <h3>Run-modes</h3>
+ * <dl>
+ *     <dt>COHORT run-mode</dt>
+ *     <dd><p>The tool can be run in the COHORT mode by using the argument <code>--run-mode COHORT</code>.
+ *      In this run-mode, coverage model parameters are inferred simultaneously with the CNV events. Depending on
+ *      available memory, it may be necessary to run the tool over a subset of all intervals, which can be specified
+ *      by -L and must be present in all of the count files. The output will contain two subdirectories, one ending with
+ *      "-model" and the other with "-calls".</p>
+ *
+ *      <p>The model subdirectory contains the inferred parameters of the coverage model, which may be used later for
+ *      CNV calling in one or more similarly-sequenced samples (see below). If a previously obtained coverage model
+ *      parameter bundle is provided via <code>--model &lt;previous_model_path&gt;</code>, those parameters will be
+ *      used for initialization and a new parameter bundle will be generated based on the provided cohort.
+ *      Furthermore, the range of genomic intervals is set to the range used for creating the previous parameter bundle
+ *      and interval-related arguments will be ignored.</p>
+ *
+ *      <p>The calls subdirectory contains one subdirectory for each sample, listing various sample-specific
+ *      quantities such as the probability of various copy-number states for each interval, the GC curve,
+ *      sample-specific unexplained variance, read depth, and loadings of various coverage bias factors.</p></dd>
+ *
+ *     <dt>CASE run-mode</dt>
+ *     <dd><p>The tool can be run in the CASE mode by using the argument <code>--run-mode CASE</code>. The path to a
+ *      previously obtained coverage model parameter bundle must be provided via <code>--model &lt;previous_model_path&gt;</code>.
+ *      The range of genomic intervals is set to the range used for creating the parameter bundle and interval-related
+ *      arguments will be ignored. The output of the CASE run-mode is only the "-calls" subdirectory.</p>
+ * </dl>
+ *
+ * <h3>Important Remarks</h3>
+ * <dl>
```

Comment:
It would be great if the usage example commands showcased parameters we expect users to commonly tweak, given certain types of data. Can we that the default parameters are meant for whole-genome sequences? What would a WES command look like?

Reply:
We're still running internal evaluations to determine the best values of parameters, many of which determine priors for the gCNV model.  We'll likely set tweak some of the default values to work well for data generated at the Broad, but as @mbabadi warns elsewhere in the documentation, there is no getting around the user having to do some experimentation to set these for their own data.

More generally, as our tools and models become more complex, we cannot expect that they'll run out of the box with one-size-fits-all parameters on all data.  Hopefully we can avoid giving this impression in documentation as well.

Reply:
Empirical example commands of what we use in production, plus a general description of the data type, would be great to showcase in the usage examples.

Reply:
@sooheelee @samuelklee While I agree with the complexity of the models and the inevitable non-universality of parameters, I am still hopeful that we can find _decent_ defaults for WGS, WGS, and gene panel data. It requires extensive evaluation on Broad and non-Broad data and is beyond the scope of the first release IMO. At least, we will try to ship the first version with non-garbage default parameters :D